src/Shared/PackageManagers/CPANProjectManager.cs

```diff
@@ -14,6 +15,9 @@ namespace Microsoft.CST.OpenSource.PackageManagers
 
     internal class CPANProjectManager : BaseProjectManager
     {
+        /// <inheritdoc cref="BaseProjectManager.Type"/>
+        public new const string Type = "cpan";
```

Comment:
Could you share some documentation as to how the const replaces the static readonly value?  I'm not familiar with this pattern.

Reply:
This is just what Rider told me to do lol. I can do some research into this pattern though.

Reply:
https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/knowing-when-to-use-override-and-new-keywords

You have to use new const to override the original const because it isn't virtual. const/static values cannot be made abstract, so this is the way to do it if you want access to the type value on a derived class.

Reply:
Since the BaseProjectManager is now abstract this could be changed by just changing the Type in that class to abstract and instead of "new const" put "override".  However, then it will be an instance variable and you'll have to create a manager to determine what types it can handle.

Reply:
I did some messing around and looks like it doesn't need to be const, I will change them to `public new readonly string Type = ...` https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/new-modifier

Reply:
If you make it readonly it means you can't access it from a static context and you will have to create a manager to determine what it's type is.  I recommend leaving it as const.

Also you'll have problems if you have, say a CpanManager but you pass it to a function that treats it like a generic manager.
<img width="1166" alt="image" src="https://user-images.githubusercontent.com/98900/161846325-0af380ee-6ce3-4851-9800-c36330ec4a10.png">

Reply:
I actually ended up changing the BaseProjectManager.Type to be a const instead of static readonly. Makes more sense this way,

Reply:
Removed `Type` from `BaseProjectManager`